lib/internal/fs/streams.js

```diff
@@ -242,10 +294,39 @@ function WriteStream(path, options) {
     options.emitClose = false;
   }
 
+  options.autoDestroy = options.autoClose === undefined ?
+    true : options.autoClose;
+  options.path = path;
   Writable.call(this, options);
+}
+Object.setPrototypeOf(WriteStream.prototype, Writable.prototype);
+Object.setPrototypeOf(WriteStream, Writable);
+
+// Backwards compat.
+Object.defineProperty(WriteStream.prototype, 'fd', {
+  get() {
+    // Backwards compat. Expects fd to be cleared before 'finish' and not
+    // through _destroy.
+    return this.autoClose && this._writableState.finished ? null : this[kFd];
```

Comment:
I feel this might create a file descriptor leak, and I would prefer not having this conditional.

Reply:
Are you fine with breaking compat? Easiest fix would be to just remove this "hack".

Reply:
How are we going to break compat by changing this?

Reply:
The previous behaviour was that `fd` is cleared before `'finish'` and not through `_destroy`. There are test that test for this.

Reply:
Let's change those tests.

Reply:
Done